TESTS/netsocket/tcp/tcpsocket_echotest.cpp

```diff
@@ -36,14 +36,23 @@ static const int pkt_sizes[PKTS] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, \
                                    };
 TCPSocket sock;
 Semaphore tx_sem(0, 1);
+events::EventQueue *event_queue;
+int bytes2recv;
+int bytes2recv_total;
 
 Timer tc_exec_time;
 int time_allotted;
+bool receive_error;
 }
 
+void tcpsocket_echotest_nonblock_receive();
+
 static void _sigio_handler(osThreadId id)
 {
     osSignalSet(id, SIGNAL_SIGIO);
+    if (event_queue != NULL) {
+        event_queue->call(tcpsocket_echotest_nonblock_receive);
+    }
 }
```

Comment:
Should here either be an else-clause or the event_queue.call() doesn't need to be wrapped on if-clause

Reply:
Again, never mind... Once you register the SIGIO handler it might be that you don't have a handle to event queue yet.

Reply:
Thanks for the remark. I changed the order of test initialization, so that the event queue gets created before we register the sigio handler.
Nevertheless, to be on the safe side, I left the NULL check and added an else statement that raises an error in case the NULL check fails :)

Reply:
ok :) This is not a suggestion to change anything but using TEST_FAIL_MESSAGE indicates there is a problem with the implementation under test. But, actually the issue would be with the test case implementation itself. MBED_ASSERT or MBED_ERROR might be good alternatives here.

Reply:
This is a very good point, thanks for that, but I think it makes some sense to stay with TEST_FAIL_MESSAGE. 
Firstly, it is not that obvious to me that the error must come from the test code. What if the handler didn't get unregistered correctly or if the EventQueue creation is erroneous?
Also, MBED_ASSERT/ERROR will crash the whole test suite and I find this reaction too serious, given the circumstances.
I hope you're ok with this? :)

Reply:
Fine by me :)